src/index.ts

```diff
@@ -66,8 +68,11 @@ switch (cmd) {
         hlFile = new HLFileScope("", argv.file);
         logErrors(hlFile);
         console.log(`Compiled "${argv.file}"`);
-        generate(hlFile);
-        if (!!!hlFile.text) { break; }
+        if(!generate(hlFile)){ 
+            if(existsSync(outPath(argv.file))){
+                unlinkSync(outPath(argv.file));// Deletes file
+            };
+            break; };
```

Comment:
This could all be added to a util
```
function cleanExistingFile(filePath: string) {
    if(!existsSync(filePath)) return;
    unlinkSync(filePath);
}
```

Which can be called like this 
```suggestion
        const filePath = outPath(argv.file);
        if (!!!hlFile.text) { 
            cleanExistingFile(filePath);
            break;
        }
```

 Its easier to follow what is happening this way I think. Adding `filePath` to a `const` variable because it is used a couple of times and doesnt need to be recalculated.

We wont have ambiguous true and false statements that way. 

IE what does it mean for `generate` to be true or false VS IF we have no text clean everything up

Reply:
Agreed, generate having a Boolean response to represent an atomic issue is ambiguous. 
A util function would be prefered.